### PR TITLE
msg/vehicle_odometry.msg: simplify covariance handling and update all usage

### PR DESCRIPTION
**Pull request by dagar**
- replace float32[21] URT covariances with smaller dedicated position/velocity/orientation covariances
 - these are easier to casually inspect and more representative of what's actually being used currently and reduces the size of vehicle_odometry_s quite a bit
 - ekf2: add new helper to get roll/pitch/yaw covariances
 - mavlink: delete unused ATT_POS_MOCAP stream (this is just a passthrough)
 - mavlink: ODOMETRY receiver
    - handle trivial frame conversions in place (MAV_FRAME_LOCAL_ENU, MAV_FRAME_LOCAL_FLU, etc)
    - ODOMETRY velocity support more than just MAV_FRAME_BODY_FRD

``` Console
pxh> listener vehicle_odometry

TOPIC: vehicle_odometry
 vehicle_odometry
    timestamp: 45748000 (0.004000 seconds ago)
    timestamp_sample: 45748000 (0 us before timestamp)
    position: [2.0012, -0.0012, 0.4814]
    q: [0.7066, 0.0020, -0.0115, 0.7075] (Roll: -0.8 deg, Pitch: -1.1 deg, Yaw: 90.1 deg)
    position_covariance: [0.0000, -0.0000, 0.0000, 0.0000, 0.0000, 0.0162]
    orientation_covariance: [0.0007, 0.0004, -0.0000, 0.0004, -0.0000, 0.0000]
    velocity: [0.0030, -0.0008, -0.0006]
    angular_velocity: [-0.0004, 0.0026, -0.0027]
    velocity_covariance: [0.0000, -0.0000, -0.0000, 0.0000, -0.0000, 0.0000]
    local_frame: 1
    velocity_frame: 1
    reset_counter: 6
    quality: 0
```

~~PR is stacked on top of https://github.com/PX4/PX4-Autopilot/pull/19965.~~

### CODE REVIEW

File: msg/vehicle_odometry.msg [hunk 1]
```diff
@@ -1,68 +1,60 @@
...
-uint8 COVARIANCE_MATRIX_YAWRATE_VARIANCE=20
-
 # Position and linear velocity frame of reference constants
-uint8 LOCAL_FRAME_NED=0         # NED earth-fixed frame
-uint8 LOCAL_FRAME_FRD=1         # FRD earth-fixed frame, arbitrary heading reference
-uint8 LOCAL_FRAME_OTHER=2       # Not aligned with the std frames of reference
-uint8 BODY_FRAME_FRD=3          # FRD body-fixed frame
+uint8 FRAME_UNKNOWN     = 0
+uint8 LOCAL_FRAME_NED   = 1 # NED earth-fixed frame
+uint8 LOCAL_FRAME_FRD   = 2 # FRD earth-fixed frame, arbitrary heading reference
```
Comment (mhkabir) [inline msg/vehicle_odometry.msg:8]: Perhaps "world-fixed" is better?

Comment (mhkabir) [inline msg/vehicle_odometry.msg:8]: "world-fixed" might be better since heading is arbitrary

File: msg/vehicle_odometry.msg [hunk 2]
```diff
@@ -1,68 +1,60 @@
...
+uint8 BODY_FRAME_FRD    = 3 # FRD body-fixed frame
```
Comment (mhkabir) [inline msg/vehicle_odometry.msg:9]: This one only makes sense for velocity right?

File: msg/vehicle_odometry.msg [hunk 3]
```diff
@@ -1,68 +1,60 @@
...
 
-# Position and linear velocity local frame of reference
+# Position frame of reference
 uint8 local_frame
```
Comment (mhkabir) [inline msg/vehicle_odometry.msg:12]:
```suggestion
uint8 position_frame
```

File: msg/vehicle_odometry.msg [hunk 4]
```diff
@@ -1,68 +1,60 @@
...
-uint8 LOCAL_FRAME_FRD=1         # FRD earth-fixed frame, arbitrary heading reference
-uint8 LOCAL_FRAME_OTHER=2       # Not aligned with the std frames of reference
-uint8 BODY_FRAME_FRD=3          # FRD body-fixed frame
+uint8 FRAME_UNKNOWN     = 0
+uint8 LOCAL_FRAME_NED   = 1 # NED earth-fixed frame
+uint8 LOCAL_FRAME_FRD   = 2 # FRD earth-fixed frame, arbitrary heading reference
+uint8 BODY_FRAME_FRD    = 3 # FRD body-fixed frame
 
-# Position and linear velocity local frame of reference
+# Position frame of reference
```
Comment (mhkabir) [inline msg/vehicle_odometry.msg:11]:
```suggestion
# Position and orientation frame of reference
```

File: msg/vehicle_odometry.msg [hunk 5]
```diff
@@ -1,68 +1,60 @@
...
+
+# Row-major representation of 3x3 position cross-covariance matrix URT.
+# Order: x, y, z
+uint8 POSITION_COVARIANCE_X_VAR  = 0
+uint8 POSITION_COVARIANCE_XY_COV = 1
+uint8 POSITION_COVARIANCE_XZ_COV = 2
+uint8 POSITION_COVARIANCE_Y_VAR  = 3
+uint8 POSITION_COVARIANCE_YZ_COV = 4
+uint8 POSITION_COVARIANCE_Z_VAR  = 5
+float32[6] position_covariance
```
[the remark this answers is not in the document]
Comment (dagar) [reply msg/vehicle_odometry.msg:27]: @james-modalai any opinion?

File: msg/vehicle_odometry.msg (hunk 2 above)
Comment (dagar) [reply msg/vehicle_odometry.msg:9]: Yes, should we split them for velocity and position? That's actually what I was already doing in EKF backend (followup PR coming).

Comment (mhkabir) [reply msg/vehicle_odometry.msg:9]: yeah probably

Comment (mhkabir) [reply msg/vehicle_odometry.msg:9]: `POSE_FRAME_NED`, etc.

**Comment by dagar** [comment]
Covariances dropped as suggested, but it's in a separate commit we can revert if someone disagrees.

File: msg/vehicle_odometry.msg [hunk 6]
```diff
@@ -1,68 +1,31 @@
...
-float32 vx 			# North velocity
-float32 vy			# East velocity
-float32 vz			# Down velocity
+float32[3] velocity         # Velocity in meters/sec. Frame of reference defined by velocity_frame variable. NaN if invalid/unknown
 
-# Angular rate in body-fixed frame (rad/s). NaN if invalid/unknown
-float32 rollspeed		# Angular velocity about X body axis
-float32 pitchspeed		# Angular velocity about Y body axis
-float32 yawspeed		# Angular velocity about Z body axis
+float32[3] angular_velocity # Angular velocity in body-fixed frame (rad/s). NaN if invalid/unknown
```
Comment (mhkabir) [inline msg/vehicle_odometry.msg:21]: This is probably unnecessary?

Comment (dagar) [reply msg/vehicle_odometry.msg:21]: Perhaps mildly useful for comparison, but otherwise we're not even using it.

Comment (mhkabir) [reply msg/vehicle_odometry.msg:21]: Most systems don't even estimate that in my experience, hence suggestion to drop

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 7]
```diff
@@ -1313,143 +1328,259 @@ MavlinkReceiver::handle_message_set_gps_global_origin(mavlink_message_t *msg)
...
+			odom.velocity[0] = odom_in.vx;
+			odom.velocity[1] = odom_in.vy;
+			odom.velocity[2] = odom_in.vz;
+			break;
+
+		case MAV_FRAME_LOCAL_ENU:
+			// ENU local tangent frame (x: East, y: North, z: Up) with origin fixed relative to earth.
+			odom.velocity_frame = vehicle_odometry_s::VELOCITY_FRAME_NED;
+			odom.velocity[0] =  odom_in.vy; // y: North
+			odom.velocity[1] =  odom_in.vx; // x: East
```
Comment (bresch) [inline src/modules/mavlink/mavlink_receiver.cpp:1481]:
```suggestion
			odom.velocity[0] =  odom_in.vy; // y: East
			odom.velocity[1] =  odom_in.vx; // x: North
```

**Review by bresch — CHANGES_REQUESTED** [review]

**Comment by mengchaoheng** [comment]
why NaN is better than 0 ?
```C++
for (auto &pc : msg.pose_covariance) {
  pc = NAN;
}
```

**Pull request merged**